robot/src/main/java/us/ilite/robot/modules/Elevator.java

```diff
@@ -0,0 +1,280 @@
+        mDesiredPower *= 0.10d; //10% of the desired power
+
+        mMasterElevator.set(Util.limit(mDesiredPower, mMinPower, mMaxPower));
+
+    }
+
+    public void updateElevator(double pNow) {
+
+        //TODO test this
+        //Should override any automated control
+        if (Math.abs(mDesiredPower) ==  0d && !mAtBottom) { //Hold if input is absent and we aren't at the bottom
+            mCurrentState = EElevatorState.HOLD;
+        } else {
+            mCurrentState = EElevatorState.NORMAL;
+        }
+
+        if (mDesiredDirectionUp) {
+            shouldDecelerate(mCurrentEncoderTicks, mDesiredDirectionUp);
+        }
+
+    }
+
+    private void resetBottom() {
+        mCurrentEncoderTicks = 0;
+        zeroEncoder();
+    }
+
+    private void resetTop() {
+        setEncoderPosition(0); //Find out top position encoder ticks
+    }
+
+    public void zeroEncoder() {
+        setEncoderPosition(0); //Actually should be zero
+    }
+
+    //Checks if we are close enough to the top or bottom 
+    //that we should be decelerating
+    private void shouldDecelerate(int pCurrentEncoderTicks, boolean pCurrentDirectionUp) {
+        if (pCurrentDirectionUp) {
+            if (pCurrentEncoderTicks >= mUpEncoderThreshold) {
+                mCurrentState = EElevatorState.DECEL_TOP;
+            }
+        } else if (!pCurrentDirectionUp) {
+            if (pCurrentEncoderTicks >= mDownEncoderThreshold) {
+                mCurrentState = EElevatorState.DECEL_BOTTOM;
+            }
+        }
+    }
+    
+    public void toBottom() {
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.SET_POSITION;
+    }
+
+    public void toTop() {
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.SET_POSITION;
+    }
+
+
+    private double setPosition(EElevatorPosition pDesiredPosition, double pCurrentTime) {
+    
+
+        double power;
+        //If the current encoder ticks is less than the threshold of the desired position, then
+        //The desired position is above the initial position
+        mDesiredPositionAboveInitial = (mCurrentEncoderTicks < pDesiredPosition.kEncoderThreshold);
+        
+        
+
+        //If the we are under the desired position and we are not in the encoder position for said
+        //position, then we are still setting position
+        if (mDesiredPositionAboveInitial && mCurrentEncoderTicks >= pDesiredPosition.kEncoderThreshold) {
+            mSettingPosition = false;
+        } else if(!mDesiredPositionAboveInitial && mCurrentEncoderTicks <= pDesiredPosition.kEncoderThreshold) { //if we are above it, same logic
+            mSettingPosition = false;
+        } else {
+            mCurrentPosition = pDesiredPosition;
+            mSettingPosition = true;
+        }
+
+        //If we make it to or beyond our position, then hold.
+        if (!mSettingPosition) {
```

Comment:
Note - the NEO brake mode is much more powerful than we're used to.  Make sure to set HOLD's power to 0 for now.